plugins/okta/komand_okta/connection/connection.py

```diff
@@ -29,3 +31,42 @@ def test(self):
             return {"success": True}
         except PluginException as error:
             raise ConnectionTestException(cause=error.cause, assistance=error.assistance, data=error.data)
+
+    def test_task(self):
+        self.logger.info("Running a connection test to Okta")
+        return_message = "The connection test to Okta was unsuccessful \n"
+        try:
+            _ = self.api_client.list_events("")
+
+            message = "The connection test to Okta was successful"
+            self.logger.info(message)
+            return {"success": True}, message
+
+        except ApiException as error:
+
+            cause_msg = f"The failed connection test to Okta was caused by: '{error.cause}'"
+            return_message += f"{cause_msg}\n"
+            self.logger.info(cause_msg)
+
+            self.logger.info(error.assistance)
+            return_message += f"{error.assistance}\n"
+
+            self.logger.error(error)
+            raise ConnectionTestException(cause=error.cause, assistance=error.assistance, data=return_message)
+
+        # This catches if the user enters a URL in the correct format but is unreachable
+        except ConnectionError as error:
+            self.logger.info(error)
+            raise ConnectionTestException(
+                cause="The URL provided in the connection is unreachable. Please ensure you are providing a valid URL when trying to connect.",
+                assistance="Please verify the URL specified is valid and reachable before attempting to connect again.",
+                data="Please ensure the URL specified can be reached and is valid before trying to reconnect.",
+            )
+
+        except ConnectionResetError as error:
+            self.logger.info("Catching egress rules for connection", error)
+            raise ConnectionTestException(
+                cause="Catching egress rules.",
+                assistance="Catching egress rules.",
```

Comment:
I don't think these are surfaced at the minute, but maybe better having something that would be an issue if we raise it?

```
cause="Failed to connect to supplied URL",
assistance="Please contact support if this issue persists"
``

Reply:
Could we update this cause and assistance field? It should save coming back to update the connection test if testing on staging proves we hit this exception and this cause/assistance is bubbled up anywhere